plotting/tests/test_plot_cache.py

```diff
@@ -0,0 +1,106 @@
+# ############################################################################### #
+#  Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+#  Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+#  SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Tests for the PlotCache class
+"""
+from unittest import TestCase
+
+from unittest.mock import patch, MagicMock, call
+
+from plotting.plot_cache import PlotCache
+
+
+# pylint:disable=protected-access
+class TestPlotCache(TestCase):
+    """Tests the functionality of the PlotCache"""
+    PLOT_PATH = "path"
+
+    def setUp(self):
+        """
+        Setup the plot cache and the opened cache file mock
+        """
+        dir_ = MagicMock()
+        params = MagicMock()
+        self.cache = PlotCache(dir_, params)
```

Comment:
This line now seems to be throwing `TypeError: expected str, bytes or os.PathLike object, not MagicMock` but I'm not sure what I changed in the rebase that broke it!